base/compiler/abstractinterpretation.jl

```diff
@@ -506,15 +506,18 @@ function add_call_backedges!(interp::AbstractInterpreter,
     @nospecialize(rettype), all_effects::Effects,
     edges::Vector{MethodInstance}, matches::Union{MethodMatches,UnionSplitMethodMatches}, @nospecialize(atype),
     sv::InferenceState)
-    # we don't need to add backedges when:
-    # - a new method couldn't refine (widen) this type and
-    # - the effects are known to not provide any useful IPO information
+    # don't bother to add backedges when both type and effects information are already
+    # maximized to the top since a new method couldn't refine or widen them anyway
     if rettype === Any
+        # ignore the `:nonoverlayed` property if `interp` doesn't use overlayed method table
+        # since it will never be tainted anyway
         if !isoverlayed(method_table(interp))
-            # we can ignore the `nonoverlayed` property if `interp` doesn't use
-            # overlayed method table at all since it will never be tainted anyway
             all_effects = Effects(all_effects; nonoverlayed=false)
         end
+        # ignore the `:noinbounds` property if it doesn't taint `:consistent`-cy
+        if !stmt_taints_inbounds_consistency(sv)
```

Comment:
Couldn't we also do this if consistency is already tainted?

Reply:
That's a good idea.